RallyHereIntegration/Source/RallyHereEditor/Private/RallyHereEditorCustomizations.cpp

```diff
@@ -0,0 +1,283 @@
+
+#include "RallyHereEditor.h"
+#include "RallyHereEditorCustomizations.h"
+
+#include "IPropertyUtilities.h"
+#include "Misc/MessageDialog.h"
+#include "Dialogs/Dialogs.h"
+#include "Dialog/SCustomDialog.h"
+#include "Widgets/DeclarativeSyntaxSupport.h"
+#include "PropertyHandle.h"
+#include "PropertyCustomizationHelpers.h"
+
+#define LOCTEXT_NAMESPACE "RallyHereEditorCustomization"
+
+void FRH_RHLegacyIdToGuidCustomization::CustomizeHeader(TSharedRef<class IPropertyHandle> StructPropertyHandle, class FDetailWidgetRow& HeaderRow, IPropertyTypeCustomizationUtils& StructCustomizationUtils)
+{
+	uint32 NumChildren;
+	StructPropertyHandle->GetNumChildren(NumChildren);
+
+	const FSimpleDelegate OnValueChanged = FSimpleDelegate::CreateLambda([&StructCustomizationUtils]()
+		{
+			StructCustomizationUtils.GetPropertyUtilities()->ForceRefresh();
+		});
+
+	for (uint32 ChildIndex = 0; ChildIndex < NumChildren; ++ChildIndex)
+	{
+		const TSharedRef< IPropertyHandle > ChildHandle = StructPropertyHandle->GetChildHandle(ChildIndex).ToSharedRef();
+
+		if (ChildHandle->GetProperty()->GetName() == TEXT("Id"))
+		{
+			IdPropertyHandle = ChildHandle;
+			ChildHandle->SetOnPropertyValueChanged(OnValueChanged);
+		}
+
+		if (ChildHandle->GetProperty()->GetName() == TEXT("LegacyId"))
+		{
+			LegacyIdPropertyHandle = ChildHandle;
+			ChildHandle->SetOnPropertyValueChanged(OnValueChanged);
+		}
+	}
+
+	check(IdPropertyHandle.IsValid());
+	check(LegacyIdPropertyHandle.IsValid());
+
+	FText Name;
+
+	TArray<void*> RawData;
+	IdPropertyHandle->AccessRawData(RawData);
+
+	FGuid Id = FGuid();
+	if (RawData.Num() > 0)
+	{
+		Id = *static_cast<FGuid*>(RawData[0]);
+	}
+
+	int32 LegacyId = 0;
+	LegacyIdPropertyHandle.Get()->GetValue(LegacyId);
+
+	TSharedRef<SHorizontalBox> ButtonArray = SNew(SHorizontalBox);
+
+	if (!Id.IsValid())
+	{
+		ButtonArray->AddSlot()
+			.AutoWidth()
+			[
+				SNew(SButton)
+				.Text(LOCTEXT("CreateItemButton", "Create"))
+				.OnClicked(this, &FRH_RHLegacyIdToGuidCustomization::OnGenerateIdClicked)
+			];
+	}
+
+	ButtonArray->AddSlot()
+	.AutoWidth()
+	[
+		SNew(SButton)
+		.Text(LOCTEXT("SetItemIdButton", "Set"))
+		.OnClicked(this, &FRH_RHLegacyIdToGuidCustomization::OnEditManuallyClicked)
+	];
+
+	HeaderRow.NameContent()
+		[
+			StructPropertyHandle->CreatePropertyNameWidget()
+		]
+	.ValueContent()
+		[
+			SNew(SVerticalBox)
+			+SVerticalBox::Slot()
+			.AutoHeight()
+			[
+				SNew(SHorizontalBox)
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(LOCTEXT("ItemId", "Item Id:"))
+				]
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(FText::FromString(FString::Printf(L"%s", *Id.ToString(EGuidFormats::DigitsWithHyphensLower))))
```

Comment:
I believe everywhere else we are dealing with guids we do not use the lowercase variant

Reply:
Ahhh, yeah might have been a auto complete failure